projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.ts

```diff
@@ -301,13 +302,23 @@ export class IgxAdvancedFilteringDialogComponent implements AfterViewInit, OnDes
         if (this.editedExpression) {
             this.editedExpression.expression.fieldName = this.selectedColumn.field;
             this.editedExpression.expression.condition = this.selectedColumn.filters.condition(this.selectedCondition);
-            this.editedExpression.expression.searchVal = this.searchValue;
+            this.editedExpression.expression.searchVal = this.transformValue(this.searchValue);
 
             this.editedExpression.inEditMode = false;
             this.editedExpression = null;
         }
     }
 
+    private transformValue(value): any {
+        if (this.selectedColumn.dataType === DataType.Number) {
+            value = parseFloat(value);
+        } else if (this.selectedColumn.dataType === DataType.Boolean) {
+            value = Boolean(value);
+        }
```

Comment:
This code is not working as expected:
```js
var myBool = Boolean("false");  // == true
```

Actually we do not support non-unary conditions for boolean columns so this code can be deleted.